tests/lib/devicetree/src/main.c

```diff
@@ -1403,16 +1403,19 @@ static void test_parent(void)
 		     "round trip through node with no compatible");
 }
 
+#undef DT_DRV_COMPAT
+#define DT_DRV_COMPAT vnd_child_bindings
 static void test_child_nodes_list(void)
 {
 	#define TEST_FUNC(child) { DT_PROP(child, val) },
 	#define TEST_MKSTR(a) _TEST_MKSTR(a)
 	#define _TEST_MKSTR(a) #a
-	#define TEST_PARENT DT_PATH(test, test_children)
-	static struct {
-		const char *v;
+	#define TEST_PARENT DT_PARENT(DT_NODELABEL(test_child_a))
+
+	struct {
+		int v;
 	} vals[] = {
-		DT_FOREACH_CHILD(TEST_PARENT, TEST_FUNC)
+		DT_INST_FOREACH_CHILD(0, TEST_FUNC)
```

Comment:
Leave this, we need a second struct or something as we want to test both DT_FOREACH_CHILD and DT_INST_FOREACH_CHILD

Reply:
Is it ok to test both in the same test, or better write 2 tests?

Reply:
Both in the same is fine with me.